src/solver/optim-model-filler/include/antares/solver/optim-model-filler/TimeDependentLinearExpression.h

```diff
@@ -44,6 +44,24 @@ class TimeDependentLinearExpression
       const Optimisation::LinearProblemApi::FillContext& fillContext,
       const LinearExpression& linearExpression);
     explicit TimeDependentLinearExpression(LinearExpressionMap linearExpressions);
+    explicit TimeDependentLinearExpression(
+      const TimeDependentLinearExpression& timeDependentLinearExpression);
+
+    // Move constructor
+    TimeDependentLinearExpression(TimeDependentLinearExpression&& other) noexcept:
+        linearExpressions_(std::move(other.linearExpressions_))
+    {
+    }
+
+    // Move assignment operator
+    TimeDependentLinearExpression& operator=(TimeDependentLinearExpression&& other) noexcept
+    {
+        if (this != &other)
+        {
+            linearExpressions_ = std::move(other.linearExpressions_);
+        }
+        return *this;
+    }
```

Comment:
This class only has a single member `linearExpressions_` and no parent class. Its move and copy constructors can be generated by the compiler, there is no need to specify them

```suggestion
    TimeDependentLinearExpression(TimeDependentLinearExpression&& other) = default;
    TimeDependentLinearExpression& operator=(TimeDependentLinearExpression&& other) = default;
```

Note: Move constructors for `std::unordered_map` are not marked `noexcept` and may throw.

Fun fact: in GCC's implementation, unordered_map's move constructor & move assignment are also compiler-generated
```cpp
      /// Move constructor.
      unordered_map(unordered_map&&) = default;
      /// Move assignment operator.
      unordered_map&
      operator=(unordered_map&&) = default;
```

Reply:
Done